x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/home/card_utils.tsx

```diff
@@ -0,0 +1,232 @@
+    release,
+    categories: ((item.categories || []) as string[]).filter((c: string) => !!c),
+    isReauthorizationRequired,
+    isUnverified,
+    isUpdateAvailable,
+    extraLabelsBadges,
+  };
+};
+
+export function getIntegrationLabels(item: PackageListItem): React.ReactNode[] {
+  const extraLabelsBadges: React.ReactNode[] = [];
+
+  if (
+    item?.installationInfo?.latest_install_failed_attempts?.some(
+      (attempt) =>
+        item.installationInfo && semverLt(item.installationInfo.version, attempt.target_version)
+    )
+  ) {
+    const upgradelFailedAttempt = item.installationInfo?.latest_install_failed_attempts?.find(
+      (attempt) =>
+        item.installationInfo && semverLt(item.installationInfo.version, attempt.target_version)
+    );
+    extraLabelsBadges.push(
+      <EuiFlexItem key="update_failed_badge" grow={false}>
+        <EuiSpacer size="xs" />
+        <span>
+          <EuiToolTip
+            title={
+              <FormattedMessage
+                id="xpack.fleet.packageCard.updateFailedTooltipTitle"
+                defaultMessage="Update failed"
+              />
+            }
+            content={upgradelFailedAttempt ? formatAttempt(upgradelFailedAttempt) : undefined}
+          >
+            <EuiBadge color="danger" iconType="error">
+              <FormattedMessage
+                id="xpack.fleet.packageCard.updateFailed"
+                defaultMessage="Update failed"
+              />
+            </EuiBadge>
+          </EuiToolTip>
+        </span>
+      </EuiFlexItem>
+    );
+  }
+
+  if (item.installationInfo?.install_status === installationStatuses.InstallFailed) {
+    const installFailedAttempt = item.installationInfo?.latest_install_failed_attempts?.find(
+      (attempt) => attempt.target_version === item.installationInfo?.version
+    );
+
+    extraLabelsBadges.push(
+      <EuiFlexItem key="install_failed_badge" grow={false}>
+        <EuiSpacer size="xs" />
+        <span>
+          <EuiToolTip
+            title={
+              <FormattedMessage
+                id="xpack.fleet.packageCard.installFailedTooltipTitle"
+                defaultMessage="Install failed"
+              />
+            }
+            content={installFailedAttempt ? formatAttempt(installFailedAttempt) : undefined}
+          >
+            <EuiBadge color="danger" iconType="error">
+              <FormattedMessage
+                id="xpack.fleet.packageCard.installFailed"
+                defaultMessage="Install failed"
+              />
+            </EuiBadge>
+          </EuiToolTip>
+        </span>
+      </EuiFlexItem>
+    );
+  }
+
+  return extraLabelsBadges;
+}
+
+function formatAttempt(attempt: InstallFailedAttempt): React.ReactNode {
+  return (
+    <>
+      <FormattedMessage
+        id="xpack.fleet.packageCard.faileAttemptDescription"
+        defaultMessage="Failed at {attemptDate}."
+        values={{
+          attemptDate: (
+            <>
+              <FormattedDate
```

Comment:
Does this handle timezone/locale automatically? I assume yes just haven't used it before so wanted to check.

Reply:
Yes it seems it will use the timezone configured in kibana (the browser one by default)